### PR TITLE
add @NonNull, @Nullable annotations to native interfaces, classes

### PR DESCRIPTION
**Pull request by dulmandakh**
This PR adds null annotations to interfaces and classes used to write native modules, that would make linting easier especially for Kotlin developers.

Test Plan:
----------
Everything will build and run as usual.

### CODE REVIEW

File: local-cli/templates/HelloWorld/android/app/build.gradle [hunk 1]
```diff
@@ -139,6 +139,7 @@ android {
 dependencies {
     compile fileTree(dir: "libs", include: ["*.jar"])
     compile "com.android.support:appcompat-v7:${rootProject.ext.supportLibVersion}"
+    compile "com.android.support:support-annotations:${rootProject.ext.supportLibVersion}"
```
Comment (gengjiawen) [inline local-cli/templates/HelloWorld/android/app/build.gradle]: I think add this dependency to template is unnecessary.

**Comment by dulmandakh** [comment]
@gengjiawen thank for the suggestion 👍

**Comment by gengjiawen** [comment]
:)

**Comment by patrickkempff** [comment]
@dulmandakh & @gengjiawen thank you both for all the android improvements lately!

File: RNTester/android/app/build.gradle [hunk 2]
```diff
@@ -142,6 +142,7 @@ android {
 dependencies {
     compile fileTree(dir: 'libs', include: ['*.jar'])
     compile 'com.android.support:appcompat-v7:26.1.0'
+    compile 'com.android.support:support-annotations:26.1.0'
```
Comment (janicduplessis) [inline RNTester/android/app/build.gradle]: Is that needed?

Comment (dulmandakh) [reply RNTester/android/app/build.gradle]: it's redundant, I'll remove it tomorrow

**Comment by dulmandakh** [comment]
@janicduplessis yep, nullable and nonnull annotations are for Kotlin support, so that it knows if a value is nullable or not.

**Comment by janicduplessis** [comment]
Alright, I removed the extra dep. Thanks for working on this!

@facebook-github-bot shipit

**Comment by dulmandakh** [comment]
@janicduplessis thank you. Could you please review https://github.com/facebook/react-native/pull/19950

**Comment by dulmandakh** [comment]
@hramos please review and merge. This will improve development experience of native module using Kotlin.

**Comment by dulmandakh** [comment]
@hramos any updates? Maybe you need to add  **react_native_dep("third-party/android/support-annotations:android-support-annotations")** as a dependency?

**Pull request closed without merging**